soh/src/code/z_en_item00.c

```diff
@@ -666,10 +672,15 @@ void func_8001E304(EnItem00* this, GlobalContext* globalCtx) {
 
 void func_8001E5C8(EnItem00* this, GlobalContext* globalCtx) {
     Player* player = GET_PLAYER(globalCtx);
+    GetItemEntry getItemEntry = Randomizer_GetRandomizedItem(this->getItemId, this->actor.id, this->ogParams, globalCtx->sceneNum);
 
     if (this->getItemId != GI_NONE) {
         if (!Actor_HasParent(&this->actor, globalCtx)) {
-            func_8002F434(&this->actor, globalCtx, this->getItemId, 50.0f, 80.0f);
+            if (!gSaveContext.n64ddFlag || getItemEntry.getItemId == GI_NONE) {
+                func_8002F434(&this->actor, globalCtx, getItemEntry.getItemId, 50.0f, 80.0f);
```

Comment:
i guess i'm just not a fan of the pattern of calling `GetRandomizedItem` when we aren't rando'd, nothing against it logically, just name wise it feels odd

Reply:
Yeah this can definitely be cleaned up a bit. I'm not actually sure why we get a pointer to the Player here either, it doesn't seem to be used anywhere.

Reply:
Resolved by 2bbd618599d9bd96250798daffbb7fbb1677db60